buildconfig/CMake/CommonSetup.cmake

```diff
@@ -412,6 +412,27 @@ if(NOT BUNDLES)
   set(BUNDLES "./")
 endif()
 
+# ##############################################################################
+# Setup pre-commit here as otherwise it will be overwritten by earlier 
+# pre-commit hooks being added
+# ##############################################################################
+option(ENABLE_PRECOMMIT "Enable pre-commit framework" ON)
+if (ENABLE_PRECOMMIT)
+  if (MSVC)
+    # Use downloaded ThirdParty version of pre-commit
+    execute_process(COMMAND bash -c "(cd ${PROJECT_SOURCE_DIR}; ${MSVC_PYTHON_EXECUTABLE_DIR}/Scripts/pre-commit.cmd install)")
+    # Create pre-commit script wrapper to use mantid third party python for pre-commit
+    execute_process(COMMAND bash -c "(cd ${PROJECT_SOURCE_DIR}; mv .git/hooks/pre-commit .git/hooks/pre-commit-script)")
+    execute_process(COMMAND bash -c "(cd ${PROJECT_SOURCE_DIR}; echo '#!/bin/sh\n${MSVC_PYTHON_EXECUTABLE_DIR}/python.exe ${PROJECT_SOURCE_DIR}/.git/hooks/pre-commit-script' >> .git/hooks/pre-commit)")
+  else()
+    # Use system installed pre-commit if not present it should just fail but 
+    # continue anyway.
+    execute_process(COMMAND bash -c "(cd ${PROJECT_SOURCE_DIR}; pre-commit install)")
+  endif()
+else()
+  message(AUTHOR_WARNING "Pre-commit not enabled by CMake, please enable manually.")
```

Comment:
I don't know under what condition this would show.

Reply:
I don't think there is as this code runs before the user has a chance to configure i.e. before clicking generate. Not sure if it is worth having the Pre_commit option?

Reply:
Just to clarify on the first run in the CMake GUI it will not trigger, but it will on the second time you click configure, or if a developer passed -DENABLE_PRECOMMIT=OFF it would show this warning.